Application/EdFi.Ods.Api/Middleware/SchoolYearRouteContextMiddleware.cs

```diff
@@ -0,0 +1,38 @@
+// SPDX-License-Identifier: Apache-2.0
+// Licensed to the Ed-Fi Alliance under one or more agreements.
+// The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+// See the LICENSE and NOTICES files in the project root for more information.
+
+using System.Threading.Tasks;
+using EdFi.Ods.Common.Context;
+using Microsoft.AspNetCore.Http;
+
+namespace EdFi.Ods.Api.Middleware
+{
+    public class SchoolYearRouteContextMiddleware : IMiddleware
+    {
+        private readonly ISchoolYearContextProvider _schoolYearContextProvider;
+
+        public SchoolYearRouteContextMiddleware(ISchoolYearContextProvider schoolYearContextProvider)
+        {
+            _schoolYearContextProvider = schoolYearContextProvider;
+        }
+
+        public async Task InvokeAsync(HttpContext context, RequestDelegate next)
+        {
+            if (context.Request.RouteValues.TryGetValue("schoolYearFromRoute", out object year))
+            {
+                if (year != null)
+                {
+                    // boxing is intentional here
```

Comment:
This is a parse operation, converting the object to a string and then to an int, not anything to do with boxing.  Probably valuable to make it clear that we're expecting it to get set as an int though.  Likely want another ticket to improve this, it should really fail out immediately if this middleware is set up and it can't extract the school year.  It makes sense for now to match the existing implementation.

Reply:
so  the tests we behaving odd, and the object could be an int or a string. but i agree another ticket makes sense in the future